yb-voyager/src/tgtdb/conn_pool_test.go

```diff
@@ -204,3 +206,62 @@ func TestUpdateConnectionsRandom(t *testing.T) {
 		assert.Equal(t, maxSize-expectedFinalSize, len(pool.idleConns))
 	}
 }
+
+func TestRemoveConnectionsForHosts(t *testing.T) {
+	// GIVEN: a conn pool of size 10.
+	size := 10
+
+	connParams := &ConnectionParams{
+		NumConnections:    size,
+		NumMaxConnections: 2 * size,
+		ConnUriList:       []string{testYugabyteDBTarget.GetConnectionString()},
+		SessionInitScript: []string{},
+	}
+	pool := NewConnectionPool(connParams)
+	assert.Equal(t, size, len(pool.conns))
+	assert.Equal(t, size, len(pool.idleConns))
+
+	host, _, err := testYugabyteDBTarget.GetHostPort()
+	testutils.FatalIfError(t, err)
+
+	//all nil case
+	pool.RemoveConnectionsForHosts([]string{host})
+
+	assert.Equal(t, size, len(pool.conns))
+	assert.Equal(t, size, len(pool.idleConns))
+
+	//4 conns to host in conns and 2 conns in idleconns
+
+	var conn *pgx.Conn
+	for i := 0; i < 4; i++ {
+		conn, _ = <-pool.conns
+		if conn == nil {
+			conn, err = pool.createNewConnection()
+			testutils.FatalIfError(t, err)
+			pool.conns <- conn
+		} else {
+			pool.conns <- conn
+		}
+	}
+
+	for i := 0; i < 2; i++ {
+		conn, _ = <-pool.idleConns
+		if conn == nil {
+			conn, err = pool.createNewConnection()
+			testutils.FatalIfError(t, err)
+			pool.idleConns <- conn
+		} else {
+			pool.idleConns <- conn
+		}
+	}
+
+	assert.Equal(t, size, len(pool.conns))
+	assert.Equal(t, size, len(pool.idleConns))
+
+	pool.RemoveConnectionsForHosts([]string{host})
+
+	assert.Equal(t, size, len(pool.conns))
+	assert.Equal(t, size, len(pool.idleConns))
```

Comment:
let's also test that all the connections in conns, and idle are nil after the call in `261`? 
Also, (if easy to do), create a connection with a different host, port, and verify that `RemoveConnectionsForHosts` does not close that connection.

Reply:
> Also, (if easy to do), create a connection with a different host, port, and verify that RemoveConnectionsForHosts does not close that connection.

I think this might be easy to do as I need a multi-node cluster for this but I am not sure if that will be easily possible with testcontainers, it will require some work.
added a todo for now, maybe will create a ticket for enhancing the yugabyteTestContainer for this functionality then we can do this.

Reply:
> Also, (if easy to do), create a connection with a different host, port, and verify that RemoveConnectionsForHosts does not close that connection.

Tried It doesn't seem to be easy, it requires some changes in the test container to either make the host configurable (as host is always localhost) or configure it to start a multi-node YB cluster. So, adding a TODO for now.

Reply:
one idea is to create a new hostname that maps to the same IP, and create a connection to that new hostname. Idk if that will work for sure, because the underlying IP would be the same, but just something worth trying. Okay to take it up later 👍